modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/introspection/EmbeddedDocExtractor.java

```diff
@@ -18,68 +18,116 @@
+                Blob content = null;
+                String name = new Path(entry.getName()).lastSegment();
+                try (InputStream is = jarFile.getInputStream(entry)) {
+                    content = getReadme(name, is);
                 }
-                if (entry.getName().startsWith(DOC_PREFIX) && !entry.isDirectory()) {
-                    String content = IOUtils.toString(is, StandardCharsets.UTF_8);
-                    String name = new Path(entry.getName()).lastSegment();
-                    if (name.length() >= 6 && name.substring(0, 6).equalsIgnoreCase("readme")) {
-
-                        ResourceDocumentationItem docItem = new ResourceDocumentationItem(name, content,
-                                DefaultDocumentationType.DESCRIPTION.toString(), bi);
-                        localDocs.put(DefaultDocumentationType.DESCRIPTION.toString(), docItem);
-                    } else {
-                        ResourceDocumentationItem docItem = new ResourceDocumentationItem(name, content,
-                                DefaultDocumentationType.HOW_TO.toString(), bi);
-                        localDocs.put(DefaultDocumentationType.HOW_TO.toString(), docItem);
-                    }
+                if (isReadme == 1) {
+                    bi.setReadme(content);
+                } else {
+                    bi.setParentReadme(content);
                 }
             }
+            if (bi.getReadme() != null && bi.getParentReadme() != null) {
+                break;
+            }
         }
-        bi.setLiveDoc(localDocs);
-        bi.setParentLiveDoc(parentDocs);
     }
+
+    /**
+     * Mimicks extaction of readme file for tests.
+     *
+     * @implNote maven is in charge of copying doc files to the target server: this needs to be done here for tests to
+     *           still take readmes into account in Eclipse for instance.
+     * @since 11.1
+     */
+    public static void extractEmbeddedDoc(File dirFile, BundleInfoImpl bi) throws IOException {
+        File readme = findFile(dirFile, README);
+        if (readme != null) {
+            try (InputStream is = new FileInputStream(readme)) {
+                bi.setReadme(getReadme(README, is));
+            }
+        }
+        File parentReadme = findFile(new File(dirFile, "../.."), README);
+        if (parentReadme != null) {
+            try (InputStream is = new FileInputStream(parentReadme)) {
+                bi.setParentReadme(getReadme(README, is));
+            }
+        }
+    }
+
+    /**
+     * Returns 0 if not a doc, 1 if local doc, 2 if parent doc.
+     */
+    protected static int isReadme(String name) {
+        boolean isDoc = name.startsWith(DOC_PREFIX);
+        boolean isParentDoc = name.startsWith(PARENT_DOC_PREFIX);
+        if (isDoc || isParentDoc) {
+            String filename = new Path(name).lastSegment();
+            if (filename.length() >= 6 && filename.substring(0, 6).equalsIgnoreCase("readme")) {
```

Comment:
This test seems a bit too optimist; as everything starting by `readme` matches. Like `readme-test.md`, `readme-sample.ascidoc`, etc.

Reply:
Indeed, i had kept the old logics but i changed it so that it better represents what's extracted.
Note that in practice "readme-sample.ascidoc" would not be retrieved by the maven plugin configuration, though, and that readme-test.md is actually accepted.

Note also that in practice, here's what we have in the nuxeo repository:

modules/runtime/nuxeo-runtime-stream/README.md
modules/runtime/nuxeo-stream/README.md
modules/runtime/nuxeo-runtime-metrics/README.md
modules/core/nuxeo-core-bulk/README.md
modules/platform/nuxeo-platform-video/README.md
modules/platform/nuxeo-platform-lang-ext/README.md
modules/platform/nuxeo-automation/nuxeo-automation-scripting/Readme.md
modules/platform/nuxeo-platform-document-routing/nuxeo-routing-default/README.md
modules/platform/nuxeo-platform-mail/README.md
modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-seqgen/README.md
modules/platform/nuxeo-elasticsearch/README.md
modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/README.md
modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-http-read-only/README.md
modules/platform/nuxeo-elasticsearch/scripts/README.md
modules/platform/nuxeo-usermapper/README.md
modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-xdocreport/ReadMe.md
modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-jod/ReadMe.md
modules/platform/nuxeo-template-rendering/ReadMe.md
modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-sandbox/ReadMe.md
modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-jaxrs/ReadMe.md
modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-jxls/ReadMe.md
modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-api/ReadMe.md
modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-core/ReadMe.md
modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-deckjs/README.md
modules/platform/nuxeo-platform-userworkspace/ReadMe.md
modules/platform/nuxeo-signature/README.md
modules/platform/nuxeo-scim-server/ReadMe.md
modules/platform/nuxeo-fsexporter/README.md
modules/platform/nuxeo-liveconnect/nuxeo-liveconnect-google-drive-core/README.md
modules/platform/nuxeo-liveconnect/nuxeo-liveconnect-dropbox-core/README.md
modules/platform/nuxeo-liveconnect/README.md
modules/platform/nuxeo-liveconnect/nuxeo-liveconnect-box-core/README.md
modules/platform/nuxeo-platform-pdf-utils/README.md
modules/platform/login/nuxeo-platform-login-openid/ReadMe.md
modules/platform/login/nuxeo-platform-login-saml2/sample/README.md
modules/platform/login/nuxeo-platform-login-saml2/README.md
modules/platform/login/nuxeo-platform-login-token/README.md
modules/platform/login/nuxeo-platform-login-keycloak/README.md
modules/platform/nuxeo-quota/README.md
modules/platform/nuxeo-platform-importer/nuxeo-importer-xml-parser/ReadMe.md
modules/platform/nuxeo-platform-importer/README.md
modules/platform/nuxeo-platform-importer/nuxeo-importer-stream/README.md
modules/platform/nuxeo-audit-storage-directory/README.md
modules/platform/nuxeo-drive-server/README.md
modules/platform/nuxeo-drive-server/nuxeo-drive-bench/README.md
modules/platform/nuxeo-apidoc-server/ReadMe.md
modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/ReadMe.md
modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/ReadMe.md
modules/platform/nuxeo-jsf-to-web-ui/README.md
modules/platform/nuxeo-duoweb-authentication/README.md
modules/platform/nuxeo-platform-3d/README.md